monai/handlers/mlflow_handler.py

```diff
@@ -198,4 +304,15 @@ def _default_iteration_log(self, engine: Engine) -> None:
         if not isinstance(loss, dict):
             loss = {self.tag_name: loss.item() if isinstance(loss, torch.Tensor) else loss}
 
-        mlflow.log_metrics(loss, step=engine.state.iteration)
+        if isinstance(engine, Trainer):
```

Comment:
I don't get why we need to check whether the engine is `Trainer` here?

Thanks.

Reply:
The check here is to avoid that users do transform on outputs of validator or evaluator which may influence the loss record.
Thanks,
Bin

Reply:
I think we don't need to take care of this logic here, user should set `iteration_log=False` for validator or evaluator: https://github.com/Project-MONAI/MONAI/blob/dev/monai/handlers/mlflow_handler.py#L83

Thanks.